modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -320,8 +320,14 @@ impl ServerModule for Wallet {
             .collect::<Vec<WalletConsensusItem>>()
             .await;
 
-        // TODO: We should not be panicking
-        let block_count = self.get_block_count().await.expect("bitcoind rpc failed");
+        // If we are unable to get a block count from the node, we default to 0 to skip
+        // adding a block count vote to consensus items.
+        //
+        // The potential impact of not including the latest block count from our peer's
+        // node is delayed processing of change outputs for the federation, which is
+        // an acceptable risk since subsequent rounds of consensus will reattempt to
+        // fetch the latest block count.
+        let block_count = self.get_block_count().await.unwrap_or(0);
```

Comment:
Can't we propose our `current_vote` again (see L339)? That seems safer.

Reply:
I've updated so we use our `current_vote` instead of 0 if we're unable to get a block count from the node, but this will still have the same effect of skipping including the block count vote to consensus items.

If we propose our `current_vote` again, we'll cause `process_consensus_item` to [`bail!`](https://github.com/fedimint/fedimint/blob/5f533fefc82355a42ac0b0c3c8cdc6da47700754/modules/fedimint-wallet-server/src/lib.rs#L382).

Calls to `process_consensus_item` appear to be idempotent when constructing a block in `complete_signed_block` using the [`unit_data_receiver`](https://github.com/fedimint/fedimint/blob/de9889bfd524f9bcb52c6e2e249701f42366a494/fedimint-server/src/consensus/server.rs#L370), however I want to flag that we will likely get unnecessary error logs since `consensus_proposal` may be called multiple times before observing a new block.

Let me know if you'd still like to propose our `current_vote` again, i.e. change

```rust
        if current_vote < block_count_proposal {
```

to

```rust
        if current_vote <= block_count_proposal {
```

Also, let me know if I'm misinterpreting your question 😃

Reply:
> If we propose our current_vote again, we'll cause process_consensus_item to [bail!](https://github.com/fedimint/fedimint/blob/5f533fefc82355a42ac0b0c3c8cdc6da47700754/modules/fedimint-wallet-server/src/lib.rs#L382).

But we aren't doing that because of the if condition further down as you noticed. I guess it would be cleaner to work with an `Option` instead? Both your original and my current solutions cause us to just not make a proposal :laughing: 

I wasn't thinking about actually on initial review, so making it explicit that we just not propose in this case would be good imo.